test-case-reporting/test/e2e.test.ts

```diff
@@ -62,20 +63,27 @@ describe.skip('end-to-end', () => {
             jobNumber: '413413.612',
             testSuiteType: 'unit',
           },
-          results: getFixture('sample-karma-report'),
+          result: getFixture('sample-karma-report'),
         });
 
+      expect(res.status).toBe(201);
+
       // TODO(#914): Replace `db('table_name').select()` calls with more readable
       // functions for getting builds, jobs, and invites.
       // See https://github.com/ampproject/amp-github-apps/blob/master/invite/src/invitation_record.ts
       // for an example of what I'm thinking of.
       const builds = await db('builds').select();
       const jobs = await db('jobs').select();
-      const testRuns = await db('testRunsLength').select();
+
+      res = await request(app).get(
+        '/test-results/build/413413?limit=100&offset=0&json=1'
+      );
+
+      const {testRuns} = res.body;
 
       expect(builds).toHaveLength(1);
       expect(jobs).toHaveLength(1);
-      expect(testRuns).toHaveLength(4);
+      expect(testRuns).toHaveLength(5);
```

Comment:
Instead of fetching builds and jobs from the DB, use the API response to assert that all the test runs have the right build `413413`, and you don't need the test itself to know the DB even exists. This is the true end-to-end you suggested would be ideal a couple weeks ago :)

Reply:
:O

Great idea! Done.